spec/rubocop/cop/lint/self_assignment_spec.rb

```diff
@@ -129,4 +129,36 @@
       foo &&= bar
     RUBY
   end
+
+  it 'registers an offense when using attribute self-assignment' do
+    expect_offense(<<~RUBY)
+      foo.bar = foo.bar
+      ^^^^^^^^^^^^^^^^^ Self-assignment detected.
+    RUBY
+  end
+
+  it 'does not register an offense when using attribute assignment' do
+    expect_no_offenses(<<~RUBY)
+      foo.bar = foo.baz
+      foo.bar
+      bar.foo = baz.foo
+      foo.bar = foo.bar + 1
```

Comment:
Can you split the `it` blocks for each testing content?

Reply:
Done for both.

Reply:
`foo.bar = foo.bar + 1` is the same as L155. Is it left out intentionally?

Reply:
Copy paste errors 🤦 Updated